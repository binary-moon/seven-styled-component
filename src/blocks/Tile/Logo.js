import styled from 'styled-components';

const Logo = styled.img`
  width: 2rem;
  height: 2rem;
  margin: 0 1rem 0 0;
`;

export default Logo;
