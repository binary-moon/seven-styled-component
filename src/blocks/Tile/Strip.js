import styled from 'styled-components';

const Strip = styled.div`
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 4rem;
  background: rgba(0, 0, 0, .7);
  display: flex;
  align-items: center;
  padding: 0 1rem;
`;

export default Strip;
