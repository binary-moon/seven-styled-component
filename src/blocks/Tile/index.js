import styled from 'styled-components';

import Image from './Image';
import Logo from './Logo';
import Strip from './Strip';

// Reusable from global H1 styles of the page
import H1 from '../../elements/H1';

const Tile = styled.div`
  position: relative;
  width: 100%;
  max-width: 30rem;
`;

Tile.Image = Image;
Tile.Logo = Logo;
Tile.Strip = Strip;
Tile.H1 = H1;

export default Tile;
