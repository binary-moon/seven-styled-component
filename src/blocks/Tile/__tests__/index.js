import React from 'react';
import Tile from '../index';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Tile block', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Tile>
        <Tile.Image src=""></Tile.Image>
        <Tile.Strip>
          <Tile.Logo src=""></Tile.Logo>
          <Tile.H1>text</Tile.H1>
        </Tile.Strip>
      </Tile>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
