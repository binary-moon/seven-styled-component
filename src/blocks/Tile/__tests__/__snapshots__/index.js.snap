// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile block renders correctly 1`] = `
.c1 {
  width: 100%;
  display: block;
}

.c3 {
  width: 2rem;
  height: 2rem;
  margin: 0 1rem 0 0;
}

.c2 {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  height: 4rem;
  background: rgba(0,0,0,.7);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
}

.c4 {
  color: white;
  font-size: 1.25rem;
  font-weight: 500;
  line-height: 2rem;
}

.c0 {
  position: relative;
  width: 100%;
  max-width: 30rem;
}

<div
  className="c0"
>
  <img
    className="c1"
    src=""
  />
  <div
    className="c2"
  >
    <img
      className="c3"
      src=""
    />
    <h1
      className="c4"
    >
      text
    </h1>
  </div>
</div>
`;
